### PR TITLE
[MB-5701] Limit Happo runs to F/E changes and master builds

### PR DESCRIPTION
**Pull request by suzubara**
## Description

In order to reduce the frequency of Happo test runs, we want to limit them to only run on PRs that have changes to F/E files _and_ on every build of `master`. This PR:
- Moves the Happo trigger to occur as a Github action instead of CircleCI
- Retain the CircleCI Storybook task because the artifacts are used to deploy Storybook, but renamed to `build_storybook`
- Use the `paths-filter` [Github action](https://github.com/marketplace/actions/paths-changes-filter) to detect whether a PR contains F/E changes. This can be tweaked if needed, but is set up to look for changes to:
  - _all_ files located in `/src` (all frontend source code)
  - Storybook config files (this can change the output of Storybook UI)
  - Happo config file (this can change the screenshots/browsers/viewports that Happo takes)
  - `yarn.lock` (dependency updates can always result in UI changes)

## Reviewer Notes

I made several commits to this PR to specifically test that the intended filters were working as expected. You can click the CI status icon next to each commit to see whether Happo was run, or view the workflow log (https://github.com/transcom/mymove/actions?query=workflow%3A%22Happo+CI%22) which will tell you what changed files were found & whether or not they triggered the frontend filter (see screenshots below).

The only thing I was unable to test is whether `push` against `master` will *always* trigger runs (whether or not F/E changes are included). I left some debug lines in the workflow just in case, but this will need to be verified once this PR is merged.

## Setup

Feel free to push/revert changes on this branch to test.. (or branch off to test but you will need to open a PR against master to trigger the workflow)
- Any PR opened against `master` that contains F/E changes should trigger Happo
- Any PR opened against `master` that does not contain F/E changes should not trigger Happo

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-5701) for this change

### Happo run is skipped, no F/E changes found:

image

image

image

### Happo run is not skipped, F/E changes found:

image

image

image

### CODE REVIEW

**Comment by suzubara** [comment]
One more change will be needed for this PR to be ✅  which is to update the branch protection settings so that `ci/circleci: storybook_tests` is not required (this will no longer ever happen)

**Review by travelar — APPROVED** [review]
Looks good. I appreciate the screenshots for visualization.

**Review by mdawn — APPROVED** [review]
Awesome! Thanks for this. 🙂

File: .circleci/config.yml [hunk 1]
```diff
@@ -873,8 +873,8 @@ jobs:
           repo: app
       - announce_failure
 
-  # `storybook_tests` triggers visual regression tests in Happo and builds static storybook assets for deploy
-  storybook_tests:
+  # `build_storybook` builds static storybook assets for deploy
+  build_storybook:
```
Comment (gidjin) [inline .circleci/config.yml:877]: The name `storybook_tests` is something our PRs look for, it's configured in the repo. You won't be able to merge until that configuration is changed to match this or this name change is undone. This is how we used to block PRs that break the tests.

Correct me if I'm wrong but I think Happo results will block the PRs without the need for this specific job name? If so I'm good removing the requirement.

File: .github/workflows/happo-tests.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          filters: |
+            frontend:
+              - 'src/**'
+              - '.storybook/**'
+              - '.happo.js'
+              - 'yarn.lock'
+
+  happo:
+    needs: changes
+    if: ${{ needs.changes.outputs.frontend == 'true' || github.ref == 'refs/heads/master' }}
```
Comment (gidjin) [inline .github/workflows/happo-tests.yml:31]: I'm tempted to try this out so we only run front end or back end unit tests when those change.

**Review by gidjin — APPROVED** [review]
Changes look good to me. Only question I had was around blocking PRs that break Happo tests. If the github workflow failing would block the PR then just let #infra know that they should disable the check for `storybook_tests`

**Review by kimallen — APPROVED** [review]
circleci/config changes approved!

File: .github/workflows/happo-tests.yml (hunk 2 above)
Comment (suzubara) [reply .github/workflows/happo-tests.yml:31]: @gidjin yes!! especially since e2e tests run for every build

**Pull request merged**